multi/imxrt-multi/cm4.c

```diff
@@ -243,17 +259,25 @@ static int runCore(unsigned int offset)
 
 static int loadFromFile(const char *path)
 {
-	char buff[256];
+	unsigned char buff[64]; /* Can't be big due to small multi stacks */
 	FILE *f;
 	size_t r, total = 0;
-	volatile char *ptr = m4_common.m4memory;
+	volatile unsigned char *ptr = m4_common.m4memory;
 
-	if ((f = fopen(path, "r")) == NULL)
+	f = fopen(path, "r");
+	if (f == NULL) {
 		return -ENOENT;
+	}
 
-	while ((r = fread(buff, sizeof(buff), 1, f)) > 0 && total < m4_common.m4memorysz) {
-		if (total + r > m4_common.m4memorysz)
+	while (total < m4_common.m4memorysz) {
+		r = fread(buff, sizeof(buff), 1, f);
+		if (r <= 0) {
```

Comment:
Currently `r < 0` is never true, `r` is declared as `size_t`, when changed to e.g. `ssize_t r;` will be ok.

Reply:
`fread()` returns `size_t`, changed condition to `r == 0`